Tests/DatadogTests/Datadog/LoggerTests.swift

```diff
@@ -551,28 +551,41 @@ class LoggerTests: XCTestCase {
         // given
         let logger = Logger.builder.build(in: core)
         Global.rum = RUMMonitor.initialize(in: core)
-        Global.rum.startView(viewController: mockView)
-        Global.rum.startUserAction(type: .tap, name: .mockAny())
         defer { Global.rum = DDNoopRUMMonitor() }
 
         // when
-        logger.info("info message")
+        Global.rum.startView(viewController: mockView)
+        logger.info("message 0")
+        Global.rum.startUserAction(type: .tap, name: .mockAny())
+        logger.info("message 1")
 
         // then
-        let logMatcher = try core.waitAndReturnLogMatchers()[0]
-        logMatcher.assertValue(
-            forKeyPath: RUMContextAttributes.IDs.applicationID,
-            equals: rum.configuration.applicationID
-        )
-        logMatcher.assertValue(
-            forKeyPath: RUMContextAttributes.IDs.sessionID,
-            isTypeOf: String.self
-        )
-        logMatcher.assertValue(
-            forKeyPath: RUMContextAttributes.IDs.viewID,
-            isTypeOf: String.self
-        )
-        logMatcher.assertValue(
+        let logMatchers = try core.waitAndReturnLogMatchers()
+        XCTAssertEqual(logMatchers.count, 2)
+
+        logMatchers.forEach {
+            $0.assertValue(
+                forKeyPath: RUMContextAttributes.IDs.applicationID,
+                equals: rum.configuration.applicationID
+            )
+        }
+        logMatchers.forEach {
+            $0.assertValue(
+                forKeyPath: RUMContextAttributes.IDs.sessionID,
+                isTypeOf: String.self
+            )
+        }
+
+        logMatchers.forEach {
+            $0.assertValue(
+                forKeyPath: RUMContextAttributes.IDs.viewID,
+                isTypeOf: String.self
+            )
+        }
+
+        logMatchers.first?.assertNoValue(forKeyPath: RUMContextAttributes.IDs.userActionID)
+
+        logMatchers.last?.assertValue(
             forKeyPath: RUMContextAttributes.IDs.userActionID,
             isTypeOf: String.self
         )
```

Comment:
trivial: `!` would be better to use here than `?` as we don't want to skip assertion when `first` and `last` are nil.
```suggestion
        logMatchers.first!.assertNoValue(forKeyPath: RUMContextAttributes.IDs.userActionID)

        logMatchers.last!.assertValue(
            forKeyPath: RUMContextAttributes.IDs.userActionID,
            isTypeOf: String.self
        )
```

Reply:
I have added the `XCTAssertEqual(logMatchers.count, 2)` as a first assertion for that 👍  I'm afraid that if we force unwrapping, we would crash the execution and prevent other tests to run :/